src/objects.c

```diff
@@ -1379,157 +1378,169 @@ void func_8000CC7C(Vehicle vehicle, u32 arg1, s32 arg2) {
-            entry->common.objectID = objectId;
-            entry->common.x = spF4[0];
-            entry->common.y = spD4[0];
-            entry->common.z = spB4[0];
-            entry->unkC = sp94[0];
-            newRacerObj = spawn_object((LevelObjectEntryCommon *) entry, OBJECT_SPAWN_UNK01);
-            newRacerObj->level_entry = NULL;
-            newRacerObj->behaviorId = BHV_TIMETRIAL_GHOST;
-            newRacerObj->shadow->scale = 0.01f;
-            newRacerObj->interactObj->flags = INTERACT_FLAGS_NONE;
-            gGhostObjStaff = newRacerObj;
-            newRacerObj->racer->transparency = 0x60;
-        }
-    }
-    gEventCountdown = 100;
-    for (j = 0; j < gNumRacers; j++) {
-        curRacerObj = (*gRacers)[j];
-        curRacer = curRacerObj->racer;
-        for (i2 = 0; i2 < 10; i2++) {
-            update_player_racer(curRacerObj, LOGIC_30FPS); // Simulate 10 updates?
+        if (gTimeTrialVehicle >= VEHICLE_BOSSES) {
+            gTimeTrialVehicle = VEHICLE_CAR;
+        }
+        // Spawn player ghost.
+        if (timetrial_valid_player_ghost()) {
+            objectID = gRacerObjectTable[gTimeTrialCharacter + (gTimeTrialVehicle * NUM_CHARACTERS)];
+            racerEntry->common.size = ((objectID & 0x100) >> 1) | 0x10;
+            racerEntry->common.objectID = objectID;
+            racerEntry->common.x = spawnX[0];
+            racerEntry->common.y = spawnY[0];
+            racerEntry->common.z = spawnZ[0];
+            racerEntry->angleY = spawnAngle[0];
+            curObj = spawn_object((LevelObjectEntryCommon *) racerEntry, 1);
+            curObj->level_entry = NULL;
+            curObj->behaviorId = BHV_TIMETRIAL_GHOST;
+            curObj->shadow->scale = 0.01f;
+            curObj->interactObj->flags = INTERACT_FLAGS_NONE;
+            gGhostObjPlayer = curObj;
+            curRacer = gGhostObjPlayer->racer;
+            curRacer->transparency = 96;
+        }
+        // Spawn staff ghost
+        if (timetrial_init_staff_ghost(get_current_map_id())) {
+            objectID = gRacerObjectTable[(gMapDefaultVehicle * NUM_CHARACTERS) + 8];
+
+            racerEntry->common.size = ((objectID & 0x100) >> 1) | 0x10;
+            racerEntry->common.objectID = objectID;
+            racerEntry->common.x = spawnX[0];
+            racerEntry->common.y = spawnY[0];
+            racerEntry->common.z = spawnZ[0];
+            racerEntry->angleY = spawnAngle[0];
+            curObj = spawn_object((LevelObjectEntryCommon *) racerEntry, 1);
+            curObj->level_entry = NULL;
+            curObj->behaviorId = BHV_TIMETRIAL_GHOST;
+            curObj->shadow->scale = 0.01f;
+            curObj->interactObj->flags = INTERACT_FLAGS_NONE;
+            gGhostObjStaff = curObj;
+            curRacer = gGhostObjStaff->racer;
+            // ASSET_OBJECT_ID_CHARACTERFLAG is what this is. But it's not used.
+            objectID = 0x40;
+            objectID += 0x20;
```

Comment:
the objectID manipulation is not needed here

Reply:
ah neat, didn't think to try without

my trust for IDO is in the negatives